File: app/src/main/java/com/epam/valkaryne/openmovieapp/domain/OpenMovieRepository.java
package com.epam.valkaryne.openmovieapp.domain;

import androidx.paging.PagingData;

import com.epam.valkaryne.openmovieapp.core.model.QueryModel;
import com.epam.valkaryne.openmovieapp.data.api.model.MovieInfo;

import io.reactivex.Observable;

public interface OpenMovieRepository {

    Observable<PagingData<MovieInfo>> getSearchResult(QueryModel query);
}

File: app/src/main/java/com/epam/valkaryne/openmovieapp/domain/OpenMovieRepository.kt
package com.epam.valkaryne.openmovieapp.domain

import androidx.paging.PagingData
import com.epam.valkaryne.openmovieapp.common.QueryModel
import com.epam.valkaryne.openmovieapp.data.api.model.MovieInfo
import kotlinx.coroutines.flow.Flow

interface OpenMovieRepository {

    fun getSearchResult(query: QueryModel): Flow<PagingData<MovieInfo>>
}